qml/AddressBarItem.qml

```diff
@@ -0,0 +1,200 @@
+                id: addressBarTextInput
+                anchors.left: parent.left
+                anchors.right: faviconImage.left
+                anchors.verticalCenter: parent.verticalCenter
+
+                clip: true
+                height: Units.gu(3.5)
+                font.family: "Prelude"
+                font.pixelSize: FontUtils.sizeToPixels("medium")
+                inputMethodHints: Qt.ImhUrlCharactersOnly
+                color: AppTweaks.privateByDefaultTweakValue ? "#2E8CF7" : "#E5E5E5"
+                selectedTextColor: "#000000"
+                selectionColor: "#ADAD15"
+                verticalAlignment: TextInput.AlignVCenter
+                horizontalAlignment: TextInput.AlignLeft
+                focus: true
+                text: ""
+
+                state: ""
+
+                onAccepted: commitURL(addressBarTextInput.text)
+                onFocusChanged: {
+                    searchSuggestions.visible = false
+                }
+
+                onActiveFocusChanged: {
+                    Qt.inputMethod.show()
+                    if (addressBarTextInput.text === "" && webViewItem.url !== "") {
+                        addressBarTextInput.text = webViewItem.url
+                    }
+                }
+
+                Component.onCompleted: {
+                    if (webViewItem.url != "") {
+                        addressBarTextInput.focus = false
+                    }
+                }
+
+                Text {
+                    anchors.fill: addressBarTextInput
+                    font.family: "Prelude"
+                    font.pixelSize: addressBarTextInput.font.pixelSize
+                    verticalAlignment: Text.AlignVCenter
+                    horizontalAlignment: Text.AlignLeft
+                    color: addressBarTextInput.color
+                    text: "Enter URL or search terms"
+                    visible: addressBarTextInput.visible && !addressBarTextInput.activeFocus && addressBarTextInput.text === ""
+                }
+
+                Binding {
+                    when: webViewItem.loading
+                    target: addressBarTextInput
+                    property: "text"
+                    value: webViewItem.url
+                }
+
+                MouseArea {
+                    anchors.fill: parent
+
+                    // here's the WebOS 3.0 text selection behavior
+                    // on clic:
+                    //  If entry is unfocused --> focus + selectAll
+                    //  clic on selected text --> show actions Copy/Cut/(Paste if clipboard non empty)
+                    //  If actions are shown --> unselect all + position cursor
+                    //  If already focused but no selection --> position cursor
+                    // on long press
+                    //  focused but no selection --> Select/SelectAll/(Paste if clipboard non empty)
+
+                    onClicked: {
+                        if (!addressBarTextInput.focus) {
```

Comment:
Currently it will not work when search suggestions are shown and you click on the address to hide the suggestions & show "cut, copy & paste".

I came up with the following instead:
                            if(addressBarTextInput.focus && addressBarTextInput.selectionStart === addressBarTextInput.selectionEnd) {
                                searchSuggestions.visible = false;
                            }

Seems to work, but you might have a more elegant solution